tests/suites/os/tests/secureboot/index.js

```diff
@@ -42,25 +53,64 @@ class secureBoot {
 	}
 
 	async testFullDiskEncryption() {
-			Promise.all(
-				[
-					{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-					{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-					{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-					{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-				].map(args => this.test.resolveMatch(
-						this.worker.executeCommandInHostOS(
-							[
-								'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-								'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-							],
-							this.link,
-						),
-						/crypt/,
-						`${args.name} is encrypted`,
-					)
+		await Promise.all(
+			[
+				{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
+				{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
+				{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
+				{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
+			].map(args => this.test.resolveMatch(
+					this.worker.executeCommandInHostOS(
+						[
+							'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
+							'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
+						],
+						this.link,
+					),
+					/crypt/,
+					`${args.name} is encrypted`,
 				)
-			);
+			)
+		);
+
+		if (!this.qmp) {
```

Comment:
This is specific to qemu/virtual devices, but the secure boot disk encryption implementation and test is common to all platforms. We could make this test generic and just abstract the machine reset, checking for the worker type and calling a QEMU or non-QEMU implementation.